lib/registerTaskWithProductionMode.js

```diff
@@ -0,0 +1,8 @@
+const productionDefaults = {
```

Comment:
We need this to actually define production tasks for styles and scripts, which are in turn invoked by the `build:production` task (see build.js).

The mechanism here does the following:
- pass arguments (in this case only one: `minify`, but potentially more …) to the production task
The mechanism to invoke prod is the same as before: `gulp styles:production`. There's actually not much change, I just removed the `sourcemaps: false` because it's false by default, and put the thing in a separate file to de-duplicate styles.js and scripts.js.

Did I address you points here?